lib/puppet/type/exec.rb

```diff
@@ -205,6 +207,16 @@ def sync
       end
     end
 
+    newparam(:use_shell, :boolean => true, :parent => Puppet::Parameter::Boolean) do
+      desc "A boolean flag that controls shell invocation.
+      By default, the provider executes commands through the standard shell.
+      It is recommended to set this parameter to `false` to execute external binaries directly"
+
+      defaultto(:true)
+
+      newvalues(:true, :false)
```

Comment:
That's my understanding as well, see ba7576d3ca6ec681301de83562ec8fd3b8207a8d. One gotcha to be aware of is puppet doesn't handle [setting the value](https://tickets.puppetlabs.com/browse/PUP-2368) or [defaulting to](https://tickets.puppetlabs.com/browse/PUP-8442) "false", because it doesn't distinguish between `false` and `nil`. Which is why we end up with `:true` and `:false` horribleness.